pypose/sparse/sparse_block_tensor.py

```diff
@@ -188,32 +189,25 @@ def coo_2_hybrid(coo, proxy):
     b_dim: List[int] = [ coo.shape[0] // s_dim[0], coo.shape[1] // s_dim[1] ] # block dimension.
 
     # Create a temporary Hybrid tensor to represent the block sequence.
-    block_seq = repeated_value_as_hybrid_value(proxy, b_dim, mode='sequence')
-    block_seq = hybrid_2_coo(block_seq).coalesce()
-
-    # Create a temporary Hybrid tensor for the placeholders.
-    block_phd = repeated_value_as_hybrid_value(proxy, b_dim, mode='zeros', dtype=coo.dtype)
-    # PyTorch may has a bug here. If one of the tensors is coalesced, the the add operation will
-    # result in a coalesced tensor, no matter whether the other tensor is coalesced or not.
-    # block_phd = hybrid_2_coo(block_phd).coalesce()
-    block_phd = hybrid_2_coo(block_phd)
-
-    # Pad the input COO tensor to have the same non-zero struture as block_seq and block_phd.
-    coo = coo + block_phd
     coo = coo.coalesce()
-
-    # Compute the indices of every element of coo inside their own respective blocks.
-    b_dim_t = torch.tensor(b_dim).to(dtype=torch.int64, device=coo.device).view((2, 1))
+    # in-block indices.
+    b_dim_t = torch.tensor(b_dim, dtype=torch.int64, device=coo.device).unsqueeze(-1)
     in_block_indices = coo.indices() % b_dim_t
+    # block indices.
+    block_indices = coo.indices() // b_dim_t
+    block_seq = torch.sparse_coo_tensor(
+            proxy.indices(), torch.arange(proxy.values().shape[0]), size=proxy.shape
+        ).to_dense()  # dense is fast in indexing
```

Comment:
The usage of `dense` here is for runtime concern. Indexing operation cost at least log(n) complexity for sparse tensor, but constant time for dense tensor. Here I prefer faster time over memory consumption.

Reply:
Sounds good. Can check the shape of proxy and decide whether to convert it to dense.

Reply:
Resolved in commit.